include/picongpu/fields/absorber/exponential/Exponential.hpp

```diff
@@ -0,0 +1,134 @@
+/* Copyright 2013-2021 Axel Huebl, Rene Widera, Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+#include "picongpu/fields/absorber/Absorber.hpp"
+#include "picongpu/fields/absorber/exponential/Exponential.kernel"
+#include "picongpu/fields/laserProfiles/profiles.hpp"
+#include "picongpu/simulation/control/MovingWindow.hpp"
+
+#include <pmacc/mappings/simulation/GridController.hpp>
+#include <pmacc/memory/dataTypes/Mask.hpp>
+#include <pmacc/traits/GetNumWorkers.hpp>
+
+#include <cstdint>
+#include <string>
+
+
+namespace picongpu
+{
+    namespace fields
+    {
+        namespace absorber
+        {
+            namespace exponential
+            {
+                /** Exponential damping field absorber
+                 *
+                 * Sets instance and implements absorption.
+                 */
+                class Exponential : public Absorber
+                {
+                public:
+                    //! Create exponential damping absorber instance
+                    Exponential()
+                    {
+                        // Copy thickness from grid.param
+                        for(uint32_t axis = 0u; axis < 3u; axis++)
```

Comment:
Is this iterating over 3 dimensions for simplification or is this required? Else I would suggest to use `axis < simDim` and update the definition too.

Could you use `axis < DIM3`. We use the define `DIM2, DIM3` where we depend on the on how many dimensions PIConGPU is supporting in the current compiled version.
This would make finding such code parts which depend on the simulation dimensionality easier if we like to refactor picongpu.

Reply:
`grid.param` requires setting all 3 as of now. I agree it makes more sense to have only till `simDim`.